main.go

```diff
@@ -339,3 +350,13 @@ func initKubeConfig(kcLocation string) *rest.Config {
 
 	return kubeConfig
 }
+
+func getOutboundIP(upstreamHost string) (string, error) {
+	conn, err := net.Dial("udp", upstreamHost)
```

Comment:
This is a really expensive operation, as the upstream host is known, can we perform this once at startup instead of on each request? Then we could also exit hard if we fail to get a local address.

Reply:
I totally agree.